src/overlays/actors/ovl_En_Bsb/z_en_bsb.c

```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
+                    }
+                }
+                func_80C0D00C(this);
+            }
+        }
+    }
+}
+
+s32 func_80C0E9CC(EnBsb* this, PlayState* play) {
+    s32 cueChannel;
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_532)) {
+        cueChannel = Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_532);
+        if (this->cueId != play->csCtx.actorCues[cueChannel]->id) {
+            if (this->unk_02B4 != 14) {
+                func_80C0E178(this);
+            }
+
+            switch (play->csCtx.actorCues[cueChannel]->id) {
+                case 1:
+                    EnBsb_ChangeAnim(this, ENBSB_ANIM_13);
+                    break;
+
+                case 2:
+                    if ((this->animIndex != ENBSB_ANIM_14) && (this->animIndex != ENBSB_ANIM_15)) {
+                        EnBsb_ChangeAnim(this, ENBSB_ANIM_14);
+                    }
+                    break;
+
+                case 3:
+                    if ((this->animIndex != ENBSB_ANIM_16) && (this->animIndex != ENBSB_ANIM_17)) {
+                        EnBsb_ChangeAnim(this, ENBSB_ANIM_16);
+                    }
+                    break;
+
+                case 4:
+                    if ((this->animIndex != ENBSB_ANIM_18) && (this->animIndex != ENBSB_ANIM_19)) {
+                        EnBsb_ChangeAnim(this, ENBSB_ANIM_18);
+                    }
+                    break;
+
+                case 5:
+                    SET_WEEKEVENTREG(WEEKEVENTREG_23_04);
+                    Actor_PlaySfx(&this->actor, NA_SE_EN_KITA_BREAK);
+                    break;
+
+                case 6:
+                    Flags_SetSwitch(play, this->switchFlag2);
+                    Actor_PlaySfx(&this->actor, NA_SE_SY_PIECE_OF_HEART);
+                    break;
+            }
+        }
+        this->cueId = play->csCtx.actorCues[cueChannel]->id;
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, cueChannel);
+        return true;
+    }
+
+    return false;
+}
+
+void EnBsb_Update(Actor* thisx, PlayState* play) {
+    EnBsb* this = THIS;
+    s32 pad;
+
+    DECR(this->unk_0292);
+
+    DECR(this->unk_0294);
+
+    DECR(this->drawDmgEffTimer);
+
+    if ((this->unk_02B4 != 0) && (this->unk_02B4 != 10) && !CHECK_WEEKEVENTREG(WEEKEVENTREG_23_04) &&
+        (this->unk_02B0 == OBJECT_BSB_LIMB_NONE)) {
+        SkelAnime_Update(&this->skelAnime);
+    }
+
+    this->actionFunc(this, play);
+    Actor_MoveWithGravity(&this->actor);
+
+    if (this->unk_02B0 == OBJECT_BSB_LIMB_NONE) {
+        func_80C0E618(this, play);
+        if ((!func_80C0E9CC(this, play)) && (this->unk_02B4 != 14)) {
+            this->actor.shape.rot.y = this->actor.world.rot.y;
+            this->actor.focus.pos.x = Math_SinS(this->actor.world.rot.y) * 20.0f + this->unk_02E0.x;
+            this->actor.focus.pos.y = this->unk_02E0.y;
+            this->actor.focus.pos.z = (Math_CosS(this->actor.world.rot.y) * 20.0f) + this->unk_02E0.z;
```

Comment:
```suggestion
            this->actor.focus.pos.x = this->unk_02E0.x + Math_SinS(this->actor.world.rot.y) * 20.0f;
            this->actor.focus.pos.y = this->unk_02E0.y;
            this->actor.focus.pos.z = this->unk_02E0.z + Math_CosS(this->actor.world.rot.y) * 20.0f;
```
note the extra set of brackets to be removed around `Math_CosS`